app/Jobs/ExtractToolsFromMetadata.php

```diff
@@ -122,14 +122,19 @@ public function toolDataset(?string $tools, int $datasetVersionId, int $userId,
 
     public function createLinkToolDatasetVersion(int $toolId, int $datasetVersionId, string $type)
     {
-        return DatasetVersionHasTool::updateOrCreate([
-            'tool_id' => $toolId,
-            'dataset_version_id' => $datasetVersionId,
-            'link_type' => $type,
-        ], [
-            'tool_id' => $toolId,
-            'dataset_version_id' => $datasetVersionId,
-            'link_type' => $type,
-        ]);
+        $check = DatasetVersionHasTool::where([
+                'tool_id' => $toolId,
+                'dataset_version_id' => $datasetVersionId,
+                'link_type' => $type,
+            ])
+            ->first();
+
+        if (is_null($check)) {
+            return DatasetVersionHasTool::create([
```

Comment:
Does this function need to return anything? It could be a `void` function. At the moment it will return `void` if the relationship exists and `int` if the relationship was created.